metron-interface/metron-rest/src/test/java/org/apache/metron/rest/controller/KafkaControllerIntegrationTest.java

```diff
@@ -54,135 +58,165 @@
 @ActiveProfiles(TEST_PROFILE)
 public class KafkaControllerIntegrationTest {
 
-    @Autowired
-    private KafkaComponent kafkaWithZKComponent;
-
-    class SampleDataRunner implements Runnable {
-
-        private boolean stop = false;
-        private String path = "../../metron-platform/metron-integration-test/src/main/sample/data/bro/raw/BroExampleOutput";
-
-        @Override
-        public void run() {
-            SampleDataGenerator broSampleDataGenerator = new SampleDataGenerator();
-            broSampleDataGenerator.setBrokerUrl(kafkaWithZKComponent.getBrokerList());
-            broSampleDataGenerator.setNum(1);
-            broSampleDataGenerator.setSelectedSensorType("bro");
-            broSampleDataGenerator.setDelay(0);
-            try {
-                while(!stop) {
-                    broSampleDataGenerator.generateSampleData(path);
-                }
-            } catch (IOException e) {
-                e.printStackTrace();
-            } catch (ParseException e) {
-                e.printStackTrace();
-            }
-        }
-
-        public void stop() {
-            stop = true;
+  private static final int KAFKA_RETRY = 10;
+  @Autowired
+  private KafkaComponent kafkaWithZKComponent;
+
+  class SampleDataRunner implements Runnable {
+
+    private boolean stop = false;
+    private String path = "../../metron-platform/metron-integration-test/src/main/sample/data/bro/raw/BroExampleOutput";
+
+    @Override
+    public void run() {
+      SampleDataGenerator broSampleDataGenerator = new SampleDataGenerator();
+      broSampleDataGenerator.setBrokerUrl(kafkaWithZKComponent.getBrokerList());
+      broSampleDataGenerator.setNum(1);
+      broSampleDataGenerator.setSelectedSensorType("bro");
+      broSampleDataGenerator.setDelay(0);
+      try {
+        while(!stop) {
+          broSampleDataGenerator.generateSampleData(path);
         }
+      } catch (IOException e) {
```

Comment:
Because you handle both `Exception`s in a same way, I think you could apply [Multiple catching](http://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html) which is available from Java 7